var/spack/repos/builtin/packages/salome-med/package.py

```diff
@@ -32,20 +38,14 @@ class SalomeMed(CMakePackage):
     variant("mpi", default=False, description="Enable MPI")
     variant("static", default=False, description="Enable static library build")
     variant("fortran", default=False, description="Enable Fortran")
+    variant("int64", default=False, description="Use 64-bit integers as indices.")
 
     depends_on("mpi", when="+mpi")
 
-    depends_on("hdf5@1.10.3+mpi", when="@4.1.0+mpi")
-    depends_on("hdf5@1.10.3~mpi", when="@4.1.0~mpi")
-
-    depends_on("hdf5@1.10.3+mpi", when="@4.0.0+mpi")
-    depends_on("hdf5@1.10.3~mpi", when="@4.0.0~mpi")
-
-    depends_on("hdf5@1.8.14+mpi", when="@3.3.1+mpi")
-    depends_on("hdf5@1.8.14~mpi", when="@3.3.1~mpi")
-
-    depends_on("hdf5@1.8.14+mpi", when="@3.2.0+mpi")
-    depends_on("hdf5@1.8.14~mpi", when="@3.2.0~mpi")
+    for _mpi_flag in ("~mpi", "+mpi"):
+        depends_on("hdf5@1.12.3{}".format(_mpi_flag), when="@5.0.0{}".format(_mpi_flag))
+        depends_on("hdf5@1.10.3{}".format(_mpi_flag), when="@4.0.0:4.1.0{}".format(_mpi_flag))
+        depends_on("hdf5@1.8.14{}".format(_mpi_flag), when="@3.2.0:3.3.1{}".format(_mpi_flag))
```

Comment:
```suggestion
    for _mpi_variant in ("~mpi", "+mpi"):
        depends_on(f"hdf5@1.12.3{_mpi_variant}", when=f"@5:{_mpi_variant}")
        depends_on(f"hdf5@1.10.3{_mpi_variant}", when=f"@4{_mpi_variant}")
        depends_on(f"hdf5@1.8.14{_mpi_variant}", when=f"@3{_mpi_variant}")
```
PS: It is officially suggested to use f-strings in `spack` now instead of `.format()`.
PPS: The `@5:` above is suggested to ensure that `hdf5` is depended on also post-`@5` (`@6:`)
PPPS: When you need to set a new hdf5 dependency version for `@6`, you'll find the place better than with the `KeyError` (like now) that would happen otherwise for `@6:`.

This proposal that you can commit using "commit suggestion" should fix most (it should fix all those) spack install errors like these:
```py
==> Error: KeyError: 'No spec with name hdf5 in salome-med@4.1.1/tsg6hep'
var/spack/repos/builtin/packages/salome-med/package.py:98, in cmake_args:
 >>     98                "-DHDF5_ROOT_DIR=%s" % spec["hdf5"].prefix,
```
This can only happen when the version ranges for depending on "hdf5" versions have holes, and 4.1.1 is in such dependency hole.